### PR TITLE
Spruce Scheduler Weather Api and Sensor model number update

### PR DESCRIPTION
**Pull request by natec007**
[213632f]Critical Update Spruce Scheduler to use new ST TWC weather api. Weather api function was partially implemented by SmartThings team, however several functions were not completed. Primary changes in weather function. Also fixed 1 bug and added version number.

[c3df129] Add updated device model number and fingerprint for Gen 3 Spruce Sensor. Update device health to 3 hours to reduce "false alarms" of inactivity.

### CODE REVIEW

File: devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy [hunk 1]
```diff
@@ -316,8 +321,8 @@ def updated(){
             sendEvent(name: 'configuration',value: 0, descriptionText: "Settings changed and will update at next report. Measure interval set to ${interval} mins")
     	}
     }
-    //check every 1 hour + 2mins
-    sendEvent(name: "checkInterval", value: 1 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    //check every 3 hours
+    sendEvent(name: "checkInterval", value: 3 * 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```
Comment (tpmanley) [inline devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy]: It looks like attribute 0x0000 on cluster 0x0405 is configured for reporting every 610 seconds ~ 10 minutes. Why the increase of checkInterval to 3 hours?

Comment (natec007) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy]: Yes, the default interval is 610 s, however, in the preferences the user can change the interval to 2 hours, so I had made it 3 to cover the longest interval.  If its preferable to keep it shorter, 2 hours would work, but would prefer to make it a little longer than the previous 1 hour, if possible.
line 342:
> //update intervals
def intervalUpdate(){
	log.debug "intervalUpdate"
    def minReport = 10
    def maxReport = 610
    if (interval != null) {
    	minReport = interval
        maxReport = interval * 61
    }
    [    
    	"zcl global send-me-a-report 0x405 0x0000 0x21 $minReport $maxReport {6400}", "delay 500",
    	"send 0x${device.deviceNetworkId} 1 1", "delay 500",
        "zcl global send-me-a-report 1 0x0000 0x21 0x0C 0 {0500}", "delay 500",
        "send 0x${device.deviceNetworkId} 1 1", "delay 500",
    ]    
}

Comment (greens) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy]: You could programmatically change the checkInterval based on that preference. I know we do that in `aeon-multisensor-6` (see the bottom of the `configure()` method). For longer `checkIntervals` we prefer `reportingInterval * 2 + 10 * 60`.

Comment (natec007) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy]: Sounds good, changed install value back to previous 62 min value, and added variable interval for when user changes poll interval.  However, did make minimum value 62 mins.  B/c some of the previous model sensors, would change the "measure" interval but would still only report after an hour.

**Review by greens — APPROVED** [review]
looks good to me

**Review by dkirker — APPROVED** [review]

**Comment by dkirker** [comment]
Merging.

@natec007 Do you have any tips for testing this? It should be in staging tomorrow, and I want to work with our QA so that they can verify this.

Also, Happy Earth Day!

**Comment by natec007** [comment]
@dkirker 
The changes are to the weather api.  Setting a schedule to run a few minutes into the future and then checking the logs will show the weather forecast and calculated weather variables when the schedule executes.  The weather will also display in the Recently.  In the Spruce Scheduler, the following fields would be set:

- Schedule Settings: Select device, set time +5mins, set today as day
-Weather Settings: Enable Seasonal Weather
-Zone Summary: Set 1 or more zones, then
- Each Zone x:  Set Sprinkler type and Landscape to anything other than off
All other values can be default
Save and that is all that is needed for test
Thanks!

**Pull request merged**